### PR TITLE
EPMRPP-89644 || Implement attachments support

### PR DESCRIPTION
**Pull request by AliakseiLiasnitski**

### CODE REVIEW

File: src/reportingApi.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import { isRPTaskMeta } from './utils';
+
+const injectRPTaskMeta = (task: Task) => {
+  if (isRPTaskMeta(task.meta)) {
+    return;
+  }
+
+  (task.meta as Models.RPTaskMeta) = {
+    ...task.meta,
+    test: {
```
Comment (AmsterGet) [inline src/reportingApi.ts]: I suppose such general name as `test` can lead to a name collision with custom user's `meta`.
Let's use more rp-specific name, f.e. `rpMeta` or smth like that.

File: src/utils.ts [hunk 2]
```diff
@@ -16,9 +16,11 @@
  */
 
 import { normalize, sep } from 'node:path';
+// eslint-disable-next-line import/named
+import { TaskMeta } from 'vitest';
```
Comment (AmsterGet) [inline src/utils.ts]: Can we import it as `import * as vitest from 'vitest';` to avoid skipping linter checks?

**Review by AmsterGet — CHANGES_REQUESTED** [review]
We also need to cover this changes in readme.

File: README.md [hunk 3]
```diff
@@ -94,3 +94,37 @@ console.error();
...
+```javascript
+import { ReportingApi } from '@reportportal/agent-js-vitest';
+```
+
+#### Reporting API methods
+
+The API provide methods for attaching data.<br/>
+
+##### attachment
+Send file to report portal for the current test. Should be called inside of corresponding test.<br/>
```
Comment (AmsterGet) [inline README.md]: We prefer using the same product naming across all its component, so let's change `report portal` to `ReportPortal` to be consistent.
```suggestion
Send file to ReportPortal for the current test. Should be called inside of corresponding test.<br/>
```

File: README.md [hunk 4]
```diff
@@ -94,3 +94,37 @@ console.error();
...
+`ReportingApi.attachment(task: vitest.Task, data: Attachment, description?: string);`<br/>
+**required**: `task`, `data`<br/>
```
Comment (AmsterGet) [inline README.md:114]: Let's describe here what the `Attachment` type is.

File: src/index.ts [hunk 5]
```diff
@@ -16,7 +16,8 @@
  */
 
 import { RPReporter } from './reporter';
+import { ReportingApi } from './reportingApi';
+import { LOG_LEVELS, STATUSES } from './constants';
```
Comment (AmsterGet) [inline src/index.ts]: Is it supposed to be used somewhere out of the agent?

File: src/models/reporting.ts [hunk 6]
```diff
@@ -14,7 +14,8 @@
  *  limitations under the License.
  *
  */
-
+// eslint-disable-next-line import/named
+import { Task, TaskMeta } from 'vitest';
```
Comment (AmsterGet) [inline src/models/reporting.ts:18]: Does it apply here as well?
> `import * as vitest from 'vitest';`

File: src/index.ts (hunk 5 above)
Comment (AliakseiLiasnitski) [reply src/index.ts]: LOG_LEVELS, STATUSES will be used for other ReportingApi methods, but now they are not needed and I will remove them

File: src/models/reporting.ts (hunk 6 above)
Comment (AliakseiLiasnitski) [reply src/models/reporting.ts:18]: I tried to apply this change but faced with lint error when `interface RPTaskMeta extends vitest.TaskMeta`
Error `'TaskMeta' not found in imported namespace 'vitest' import/namespace`

File: README.md [hunk 7]
```diff
@@ -94,3 +94,38 @@ console.error();
...
+
+  ReportingApi.attachment(task, {
+    name: fileName,
+    type: 'image/png',
+    content: fileContent.toString('base64'),
+  }, 'Description');
+
+  expect(true).toBe(true);
+});
+```
```
Comment (AmsterGet) [inline README.md:131]: Please also describe how to use `setup` file and how the `ReportingApi` usage will be changed with that.

**Review by AmsterGet — APPROVED** [review]

**Pull request merged**